### ISSUE TITLE
DOKKU_HOST_ROOT is ignored for plugin data

### ISSUE THREAD

**Issue by salomvary** [dokku/dokku#5468]
### Description of problem

I'm using the [Docker installation method](https://dokku.com/docs/getting-started/install/docker/) which went well until I installed the PostgreSQL plugin and created a service. I noticed, that while most Dokku data is stored in the configured custom `DOKKU_HOST_ROOT` folder that is _not_ `/var/lib/dokku`, the PostgreSQL plugin created it's data under `/var/lib/dokku/services/postgres/`.

### Steps to reproduce

On the host (substitute `/path/to/dokku` with anything that's not `/var/lib/dokku`):

```
docker container run \
  --env DOKKU_HOSTNAME=dokku.me \
  --env DOKKU_HOST_ROOT=/path/to/dokku/home/dokku \
  --name dokku \
  --publish 3022:22 \
  --publish 8080:80 \
  --publish 8443:443 \
  --volume /path/to/dokku:/mnt/dokku \
  --volume /var/run/docker.sock:/var/run/docker.sock \
  dokku/dokku:0.28.4
```

On Dokku:
```
dokku plugin:install https://github.com/dokku/dokku-postgres.git
dokku apps:create python-getting-started
dokku postgres:create python-getting-started-database
dokku postgres:link python-getting-started-database python-getting-started
```

On the host:
```
ls /path/to/dokku
# Lists expected stuff

ls /var/lib/dokku
python-getting-started-database
# Should not even exist, but it does and contains PostgreSQL data
```

### dokku report $APP_NAME

```
dokku report python-getting-started
-----> uname: Linux dokku.lan 5.10.60-qnap #1 SMP Tue May 31 07:32:23 CST 2022 x86_64 x86_64 x86_64 GNU/Linux
-----> memory: 
                     total        used        free      shared  buff/cache   available
       Mem:           3778        2390         353         121        1034         803
       Swap:         23967        1702       22264
-----> docker version: 
       Client:
        Version:           20.10.12
        API version:       1.41
        Go version:        go1.16.2
        Git commit:        20.10.12-0ubuntu2~20.04.1
        Built:             Wed Apr  6 02:14:38 2022
        OS/Arch:           linux/amd64
        Context:           default
        Experimental:      true
       
       Server: Docker Engine - Community
        Engine:
         Version:          20.10.14-qnap2
         API version:      1.41 (minimum version 1.12)
         Go version:       go1.16.15
         Git commit:       3bb6db2
         Built:            Fri Apr  8 10:09:48 2022
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          v1.5.11
         GitCommit:        3df54a852345ae127d1fa3092b95168e4a88e2f8
        runc:
         Version:          1.0.3
         GitCommit:        v1.0.3-0-gf46b6ba2
        docker-init:
         Version:          0.19.0
         GitCommit:        de40ad0
-----> docker daemon info: 
       Client:
        Context:    default
        Debug Mode: true
       
       Server:
        Containers: 7
         Running: 7
         Paused: 0
         Stopped: 0
        Images: 45
        Server Version: 20.10.14-qnap2
        Storage Driver: overlay2
         Backing Filesystem: extfs
         Supports d_type: true
         Native Overlay Diff: false
         userxattr: false
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Cgroup Version: 1
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay qnet
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux kata-runtime runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: 3df54a852345ae127d1fa3092b95168e4a88e2f8
        runc version: v1.0.3-0-gf46b6ba2
        init version: de40ad0
        Security Options:
         seccomp
          Profile: default
        Kernel Version: 5.10.60-qnap
        Operating System: QTS 5.0.0 (20220531)
        OSType: linux
        Architecture: x86_64
        CPUs: 2
        Total Memory: 3.69GiB
        Name: NAS
        ID: 3GNM:GOSJ:VWE3:2MD6:AU3Z:D3P2:ZR2T:SSIH:PEZR:KVS6:ICJA:IDQY
        Docker Root Dir: /share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker
        Debug Mode: true
         File Descriptors: 26697
         Goroutines: 119
         System Time: 2022-11-18T22:56:59.303201993+01:00
         EventsListeners: 1
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false
        Product License: Community Engine
        Default Address Pools:
          Base: 172.29.0.0/16, Size: 22
       
-----> git version: git version 2.25.1
-----> sigil version: 0.9.0build+bc921b7
-----> herokuish version: 
       herokuish: 0.5.39
       buildpacks:
         heroku-buildpack-multi     v1.2.0
         heroku-buildpack-ruby      v244
         heroku-buildpack-nodejs    v202
         heroku-buildpack-clojure   v90
         heroku-buildpack-python    v222
         heroku-buildpack-java      v72
         heroku-buildpack-gradle    v38
         heroku-buildpack-scala     v96
         heroku-buildpack-play      v26
         heroku-buildpack-php       v227
         heroku-buildpack-go        v168
         heroku-buildpack-nginx     v22
         buildpack-null             v3
-----> dokku version: dokku version 0.28.4
-----> plugn version: plugn: 0.12.0build+3a27594
-----> dokku plugins: 
         00_dokku-standard    0.28.4 enabled    dokku core standard plugin
         20_events            0.28.4 enabled    dokku core events logging plugin
         app-json             0.28.4 enabled    dokku core app-json plugin
         apps                 0.28.4 enabled    dokku core apps plugin
         builder              0.28.4 enabled    dokku core builder plugin
         builder-dockerfile   0.28.4 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.28.4 enabled    dokku core builder-herokuish plugin
         builder-lambda       0.28.4 enabled    dokku core builder-lambda plugin
         builder-null         0.28.4 enabled    dokku core builder-null plugin
         builder-pack         0.28.4 enabled    dokku core builder-pack plugin
         buildpacks           0.28.4 enabled    dokku core buildpacks plugin
         caddy-vhosts         0.28.4 enabled    dokku core caddy-vhosts plugin
         certs                0.28.4 enabled    dokku core certificate management plugin
         checks               0.28.4 enabled    dokku core checks plugin
         common               0.28.4 enabled    dokku core common plugin
         config               0.28.4 enabled    dokku core config plugin
         cron                 0.28.4 enabled    dokku core cron plugin
         docker-options       0.28.4 enabled    dokku core docker-options plugin
         domains              0.28.4 enabled    dokku core domains plugin
         enter                0.28.4 enabled    dokku core enter plugin
         git                  0.28.4 enabled    dokku core git plugin
         logs                 0.28.4 enabled    dokku core logs plugin
         network              0.28.4 enabled    dokku core network plugin
         nginx-vhosts         0.28.4 enabled    dokku core nginx-vhosts plugin
         plugin               0.28.4 enabled    dokku core plugin plugin
         postgres             1.25.0 enabled    dokku postgres service plugin
         proxy                0.28.4 enabled    dokku core proxy plugin
         ps                   0.28.4 enabled    dokku core ps plugin
         registry             0.28.4 enabled    dokku core registry plugin
         repo                 0.28.4 enabled    dokku core repo plugin
         resource             0.28.4 enabled    dokku core resource plugin
         run                  0.28.4 enabled    dokku core run plugin
         scheduler            0.28.4 enabled    dokku core scheduler plugin
         scheduler-docker-local 0.28.4 enabled    dokku core scheduler-docker-local plugin
         scheduler-null       0.28.4 enabled    dokku core scheduler-null plugin
         shell                0.28.4 enabled    dokku core shell plugin
         ssh-keys             0.28.4 enabled    dokku core ssh-keys plugin
         storage              0.28.4 enabled    dokku core storage plugin
         trace                0.28.4 enabled    dokku core trace plugin
         traefik-vhosts       0.28.4 enabled    dokku core traefik-vhosts plugin
=====> python-getting-started app-json information
       App json computed selected:    app.json
       App json global selected:      app.json
       App json selected:             
=====> python-getting-started app information
       App created at:                1668806559
       App deploy source:             python-getting-started
       App deploy source metadata:    python-getting-started
       App dir:                       /home/dokku/python-getting-started
       App locked:                    false
=====> python-getting-started builder information
       Builder build dir:             
       Builder computed build dir:    
       Builder computed selected:     
       Builder global build dir:      
       Builder global selected:       
       Builder selected:              
=====> python-getting-started builder-dockerfile information
       Builder dockerfile computed dockerfile path: Dockerfile               
       Builder dockerfile global dockerfile path: Dockerfile               
       Builder dockerfile dockerfile path:                          
=====> python-getting-started builder-lambda information
       Builder lambda computed lambdayml path: lambda.yml               
       Builder lambda global lambdayml path: lambda.yml               
       Builder lambda lambdayml path:                          
=====> python-getting-started builder-pack information
       Builder pack computed projecttoml path: project.toml             
       Builder pack global projecttoml path: project.toml             
       Builder pack projecttoml path:                          
=====> python-getting-started buildpacks information
       Buildpacks computed stack:     gliderlabs/herokuish:latest-20
       Buildpacks global stack:       
       Buildpacks list:               
       Buildpacks stack:              
=====> python-getting-started ssl information
       Ssl dir:                       /home/dokku/python-getting-started/tls
       Ssl enabled:                   false                    
       Ssl hostnames:                                          
       Ssl expires at:                                         
       Ssl issuer:                                             
       Ssl starts at:                                          
       Ssl subject:                                            
       Ssl verified:                                           
=====> python-getting-started checks information
       Checks disabled list:          none                     
       Checks skipped list:           none                     
=====> python-getting-started cron information
       Cron task count:               0
=====> python-getting-started docker options information
       Docker options build:          --link dokku.postgres.python-getting-started-database:dokku-postgres-python-getting-started-database 
       Docker options deploy:         --link dokku.postgres.python-getting-started-database:dokku-postgres-python-getting-started-database --restart=on-failure:10 
       Docker options run:            --link dokku.postgres.python-getting-started-database:dokku-postgres-python-getting-started-database 
=====> python-getting-started domains information
       Domains app enabled:           true                     
       Domains app vhosts:            python-getting-started.dokku.lan
       Domains global enabled:        true                     
       Domains global vhosts:         dokku.lan                
=====> python-getting-started git information
       Git deploy branch:             master                   
       Git global deploy branch:      master                   
       Git keep git dir:              false                    
       Git rev env var:               GIT_REV                  
       Git sha:                       ca790e6                  
       Git last updated at:           1668806559               
=====> python-getting-started logs information
       Logs computed max size:        10m
       Logs global max size:          10m
       Logs global vector sink:       
       Logs max size:                 
       Logs vector sink:              
=====> python-getting-started network information
       Network attach post create:           
       Network attach post deploy:           
       Network bind all interfaces:          false
       Network computed attach post create:  
       Network computed attach post deploy:  
       Network computed bind all interfaces: false
       Network computed initial network:     
       Network computed tld:                 
       Network global attach post create:    
       Network global attach post deploy:    
       Network global bind all interfaces:   false
       Network global initial network:       
       Network global tld:                   
       Network initial network:              
       Network static web listener:          
       Network tld:                          
       Network web listeners:                10.0.3.5:5000
=====> python-getting-started nginx information
       Nginx access log format:                                
       Nginx access log path:         /var/log/nginx/python-getting-started-access.log
       Nginx bind address ipv4:                                
       Nginx bind address ipv6:       ::                       
       Nginx client max body size:                             
       Nginx disable custom config:   false                    
       Nginx error log path:          /var/log/nginx/python-getting-started-error.log
       Nginx global hsts:             true                     
       Nginx computed hsts:           true                     
       Nginx hsts:                                             
       Nginx hsts include subdomains: true                     
       Nginx hsts max age:            15724800                 
       Nginx hsts preload:            false                    
       Nginx proxy buffer size:       4096                     
       Nginx proxy buffering:         on                       
       Nginx proxy buffers:           8 4096                   
       Nginx proxy busy buffers size: 8192                     
       Nginx proxy read timeout:      60s                      
       Nginx last visited at:         1668806643               
       Nginx x forwarded for value:   $remote_addr             
       Nginx x forwarded port value:  $server_port             
       Nginx x forwarded proto value: $scheme                  
       Nginx x forwarded ssl:                                  
=====> python-getting-started proxy information
       Proxy enabled:                 true
       Proxy port map:                http:80:5000
       Proxy type:                    nginx
=====> python-getting-started ps information
       Deployed:                      true
       Processes:                     1
       Ps can scale:                  true
       Ps computed procfile path:     Procfile
       Ps global procfile path:       Procfile
       Ps procfile path:              
       Ps restart policy:             on-failure:10
       Restore:                       true
       Running:                       true
       Status web 1:                  running (CID: 1cc6486ffc4)
=====> python-getting-started registry information
       Registry computed image repo:      dokku/python-getting-started
       Registry computed push on release: false
       Registry computed server:          
       Registry global push on release:   
       Registry global server:            
       Registry image repo:               
       Registry push on release:          
       Registry server:                   
       Registry tag version:              
=====> python-getting-started resource information
=====> python-getting-started scheduler information
       Scheduler computed selected:   docker-local
       Scheduler global selected:     docker-local
       Scheduler selected:            
=====> python-getting-started scheduler-docker-local information
       Scheduler docker local disable chown:                          
       Scheduler docker local init process: true                     
       Scheduler docker local parallel schedule count:                          
=====> python-getting-started storage information
       Storage build mounts:                                   
       Storage deploy mounts:                                  
       Storage run mounts: 
```

### Additional information

Maybe I'm wrong with interpreting what `DOKKU_HOST_ROOT` might mean and this is the expected behavior? But it would be great to have the option to not write anything to `/var/lib/dokku`. My current workaround is to have a symlink from `/var/lib/dokku` to the desired folder.

Thanks for creating Dokku by the way, it's an amazing project!

```
docker inspect dokku.postgres.python-getting-started-database
[
    {
        "Id": "1444aab90dbc66f1e37106971900998986d2a06153f1208332e6e7239d34e593",
        "Created": "2022-11-18T20:54:27.934914273Z",
        "Path": "docker-entrypoint.sh",
        "Args": [
            "postgres"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 28830,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2022-11-18T20:54:54.621515978Z",
            "FinishedAt": "2022-11-18T20:54:45.025058941Z"
        },
        "Image": "sha256:cefd1c9e490c8b581d834d878081cf64c133df1f9f443c5e5f8d94fbd7c7a1d4",
        "ResolvConfPath": "/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/containers/1444aab90dbc66f1e37106971900998986d2a06153f1208332e6e7239d34e593/resolv.conf",
        "HostnamePath": "/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/containers/1444aab90dbc66f1e37106971900998986d2a06153f1208332e6e7239d34e593/hostname",
        "HostsPath": "/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/containers/1444aab90dbc66f1e37106971900998986d2a06153f1208332e6e7239d34e593/hosts",
        "LogPath": "/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/containers/1444aab90dbc66f1e37106971900998986d2a06153f1208332e6e7239d34e593/1444aab90dbc66f1e37106971900998986d2a06153f1208332e6e7239d34e593-json.log",
        "Name": "/dokku.postgres.python-getting-started-database",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "/var/lib/dokku/services/postgres/python-getting-started-database/data:/var/lib/postgresql/data"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {
                    "max-file": "10",
                    "max-size": "10m"
                }
            },
            "NetworkMode": "default",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "always",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "host",
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": [
                {
                    "Name": "nofile",
                    "Hard": 65535,
                    "Soft": 65535
                }
            ],
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/b756790f1112dd5a432e47f49460d52f9c304b39c3c5b0c0d327ac65c44b448d-init/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/14b7e2cc28b8a7874464c6d238962759e522efde5474e1e959e7244fe301e4c4/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/adcc2ceff4d56cba0295a5b2dcb3faaac0c3b4a1c1d4d4f6f353c155706ecdf3/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/6aed11f323f0609e971b7709436a40f26f741cead15b33006dbda4a5cc02234e/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/253d0837c95f5291f7a823812611c553270cf00c4c766b78fc0560a41f664c89/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/d55277495a6cabaa98937d1f771f97597818f83ec6cbf1dcb99632cb16264de9/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/5d3b00a1a2f781df2e8df17170f4ef90cee2425bb4ffeb2294703b8508248dc8/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/61462b78587a8d219eee3ebb4c0abf5ac1da08454e816dd51c020fb5db40d09b/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/6162a8e97d82645283d625092839598c23132164999bbf58a376bcf53d9886ec/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/41f6e81a5d4106afe8620b93aa4c414a5ec5263edfa82ed7ce6c5fb086cdeb71/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/f2ab345972436f1429f555c602bf736e77e85d9ca7957d14ef64dc2ce75b7e9b/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/4a837bbc25f4a270a85217827c4c2014dc1ea2048da8d93787c4f61e9488a4c7/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/3c05b500f4c3838d5c4090e4048ae0511ae690030dccfd4b37ca73da19786d30/diff:/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/0d6b129debcc36f8edcb9c5586c11a6774ab963dd02fb9972ad2b7a1aff850fe/diff",
                "MergedDir": "/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/b756790f1112dd5a432e47f49460d52f9c304b39c3c5b0c0d327ac65c44b448d/merged",
                "UpperDir": "/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/b756790f1112dd5a432e47f49460d52f9c304b39c3c5b0c0d327ac65c44b448d/diff",
                "WorkDir": "/share/CE_CACHEDEV1_DATA/Container/container-station-data/lib/docker/overlay2/b756790f1112dd5a432e47f49460d52f9c304b39c3c5b0c0d327ac65c44b448d/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/var/lib/dokku/services/postgres/python-getting-started-database/data",
                "Destination": "/var/lib/postgresql/data",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "1444aab90dbc",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "5432/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "POSTGRES_PASSWORD=e854e143b73bfb0b32537b1c4604ed88",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/14/bin",
                "GOSU_VERSION=1.14",
                "LANG=en_US.utf8",
                "PG_MAJOR=14",
                "PG_VERSION=14.5-2.pgdg110+2",
                "PGDATA=/var/lib/postgresql/data"
            ],
            "Cmd": [
                "postgres"
            ],
            "Image": "postgres:14.5",
            "Volumes": {
                "/var/lib/postgresql/data": {}
            },
            "WorkingDir": "",
            "Entrypoint": [
                "docker-entrypoint.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "dokku": "service",
                "dokku.service": "postgres"
            },
            "StopSignal": "SIGINT"
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "07fdefdc864cd3af77fbc0932b60d6bbcaa3b42d5b42ed917652798796291830",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "5432/tcp": null
            },
            "SandboxKey": "/var/run/docker/netns/07fdefdc864c",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "955d6adefb768058bd497fdf44b4035528703539251d8a698dfa1fb83d8c1ede",
            "Gateway": "10.0.3.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "10.0.3.3",
            "IPPrefixLen": 24,
            "IPv6Gateway": "",
            "MacAddress": "02:42:0a:00:03:03",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "3d658d7c47a1806404be284af4ca7310344f17044a428bd5013946e7d7ce51da",
                    "EndpointID": "955d6adefb768058bd497fdf44b4035528703539251d8a698dfa1fb83d8c1ede",
                    "Gateway": "10.0.3.1",
                    "IPAddress": "10.0.3.3",
                    "IPPrefixLen": 24,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:0a:00:03:03",
                    "DriverOpts": null
                }
            }
        }
    }
]
```

**Comment by salomvary** [comment]
@josegonzalez Thanks for the response, I may or may not agree with you depending whether you mean `/var/lib/dokku/services/postgres/` should be on the _host_ file system or the _container_ one :)

Maybe I've misunderstood what `DOKKU_HOST_ROOT` is supposed to configure (found no documentation on the exact meaning of this environment variable).

Let me rephrase this from the perspective of the concrete problem I was having. I have a host system where `/var/lib/` is not supposed to be written to (it's a ramdisk with very limited space, wiped on each reboot) therefore all Dokku data written on the host file system should be configurable, including plugins data.

These two lines in the documentation suggested that if I change `/var/lib/dokku` to `/not/var/lib/dokku` on the host side of the `docker container run` command, nothing will be written to `/var/lib/dokku` but that is not the case.

```
  --env DOKKU_HOST_ROOT=/var/lib/dokku/home/dokku \
  --volume /var/lib/dokku:/mnt/dokku \
```

So maybe it's not a but concerning `DOKKU_HOST_ROOT` but a feature request to add `DOKKU_HOST_LIB_ROOT` or something like that?

**Comment by josegonzalez** [comment]
In your example there, you aren't actually setting anything to `/not/var/lib/dokku`.

The container path `/var/lib/dokku` is always mounted from `/mnt/dokku`.  So you can mount whatever you want there.

Say you want the host path to be something like `/data/dokku. The docker run command would be:

```shell
docker container run \
  --env DOKKU_HOSTNAME=dokku.me \
  --env DOKKU_HOST_ROOT=/data/home/dokku \
  --name dokku \
  --publish 3022:22 \
  --publish 8080:80 \
  --publish 8443:443 \
  --volume /data/dokku:/mnt/dokku \
  --volume /var/run/docker.sock:/var/run/docker.sock \
  dokku/dokku:0.28.4
```

Each datastore stores their data in `/var/lib/dokku/services/$DATASTORE`. That path is controlled on a datastore-dependent env var, so `POSTGRES_HOST_ROOT` for postgres. The code for that is [here](https://github.com/dokku/dokku-postgres/blob/d281d63afb930ae47db19b6e27dd27aa72a5d191/config#L5-L6). To override that with the above example, you'd probably want:

```shell
docker container run \
  --env DOKKU_HOSTNAME=dokku.me \
  --env DOKKU_HOST_ROOT=/data/home/dokku \
  --env POSTGRES_HOST_ROOT=/data/home/services/postgres \
  --name dokku \
  --publish 3022:22 \
  --publish 8080:80 \
  --publish 8443:443 \
  --volume /data/dokku:/mnt/dokku \
  --volume /var/run/docker.sock:/var/run/docker.sock \
  dokku/dokku:0.28.4
```

**Comment by salomvary** [comment]
> In your example there, you aren't actually setting anything to `/not/var/lib/dokku`.

Sorry, now that I re-read my last message above, the example is misleading. What I meant to say is:

 These two lines  [in the documentation](https://dokku.com/docs/getting-started/install/docker/):
 
 ```
   --env DOKKU_HOST_ROOT=/var/lib/dokku/home/dokku \
    --volume /var/lib/dokku:/mnt/dokku \
```

...and this sentence: "Application repositories, plugin config, as well as **plugin data are persisted to disk within the specified host directory** for /var/lib/dokku."


...suggested that if I change `/var/lib/dokku` to `/not/var/lib/dokku` on the host side of the `docker container run` command, like this:

 ```
   --env DOKKU_HOST_ROOT=/not/var/lib/dokku/home/dokku \
    --volume /not/var/lib/dokku:/mnt/dokku \
```

nothing will be written to `/var/lib/dokku`.

But reading your last reply here you suggest that `DOKKU_HOST_ROOT` does not apply to plugins and I need to set `POSTGRES_HOST_ROOT`.

**Comment by josegonzalez** [comment]
Okay I think I understand where you are coming from.

Do you mind checking if my instructions re:Postgres work for putting the data where you expect? We can modify the plugins to do the right thing if they do.

**Comment by josegonzalez** [comment]
Okay if you have:

```
--env DOKKU_HOST_ROOT=/Users/josediazgonzalez/go/src/github.com/dokku/5468-test/home/dokku
```

And

```
--env POSTGRES_HOST_ROOT=/Users/josediazgonzalez/go/src/github.com/dokku/5468-test/services/postgres
```

The data is persisted to disk as expected. You need both, otherwise dokku will write data in whatever is `$DOKKU_LIB_ROOT/services/$SERVICE_TYPE`, and then that isnt exposed inside the container, so some of the data (the service data) gets written on the host, and other data (config files and the like) get written in the container.

I think modifying the plugins to use something like `$DOKKU_LIB_HOST_ROOT` will work, though will of course require users to update their plugins (the omission of that variable will result in the plugins working as they currently do).

**Comment by josegonzalez** [comment]
Ah one thing I'm seeing with that change is that other than the mounted files, everything else seems to not be on the host... I guess I'm seeing what you are seeing. Interesting.

**Comment by josegonzalez** [comment]
Ah I see.

We add a magic symlink for `config` and `data` in the `Dockerfile`. Without these, we wouldn't respect the mount for `/var/lib/dokku/config` and `/var/lib/dokku/data`. As such, the fix is not bc-compatible, since we need a fix both in the plugins as well as in dokku (and the latter might break installations for folks).

We can fix this, but will need to be for the next minor - probably in February.

As an alternative, one could pre-create the `/var/lib/dokku/services` directory in the dir that gets mounted and then add that as a volume mount, which would effectively be the same as adding a symlink.

**Comment by josegonzalez** [comment]
Okay so the above should help with the workaround, and will also allow us to proceed with the actual fix in the next minor.

**Comment by maciejopalinski** [comment]
@josegonzalez 
Any updates? Or even a description of a workaround? I really urgently need it, because the most basic plugins don't work...
